action_type.cpp

```diff
@@ -32,6 +36,9 @@ std::ostream &operator<<(std::ostream &os, const action_type t)
 
 std::string to_str(const action_type this_action_type)
 {
+  #ifdef HAS_MAGIC_ENUM // HAS_MAGIC_ENUM is defined in game.pro
+  return std::string(magic_enum::enum_name(this_action_type));
+  #else

```

Comment:
As said above, I do enjoy that the user gets the option to use magic_enum or not without breaking the build, but then I wonder about the usefulness of using magic_enum at all? If we are not enforcing the  use of the library, then we are just adding more lines to a code that was already functional?

Is this solution transitory? I.e., do you intend to eventually enforce use of magic_enum?

Reply:
@TheoPannetier I completely agree. This solution is transitory and I do intend to eventually enforce the use of `magic_enum`. Luckily, I have my `HAS_MAGIC_ENUM` already in place :-)